Shipping Labels: populated the total weight automatically in packages step

pmusolino opened the pull request:
Fixes #4090 by fetching and populating the weight of the package for every item.

## Testing
1. Modify products and product variations by specifying different weights.
2. Create an order using these products, and make sure it satisfies the SL creation conditions.
3. Open the order in the app.
4. Click on Create Shipping Label.
5. Pass the origin and shipping addresses steps.
6. Open the package details screen.
7. Confirm that the shown "Total package weight" is the correct one.

## Screenshot
<img src="https://user-images.githubusercontent.com/495617/120478002-ae3da980-c3ac-11eb-9e92-7ef62c267cf6.png" width=300 />


Update release notes:
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

pmusolino committed:
feat: populate the weight automatically in packages step

rachelmcr reviewed (commented):
This looks good and tests well with different products (regular product with weight, variation with weight, product without a set weight) as long as I follow the expected user flow. However, I found a few cases that we might want to consider:

1. When none of the items in the order have a set weight, or the user manually enters `0` as the weight, the Done button is still enabled and I can save those package details. However, in WP Admin the total weight field doesn't allow this:
![Screen Shot 2021-06-03 at 11 21 57 AM](https://user-images.githubusercontent.com/8658164/120629671-f668d480-c45d-11eb-8549-ee6cafdcf4ca.png)

2. If I manually update the total weight, and tap Done, I see that the weight I entered is saved. However, if I tap the Package Details cell again we use `setTotalWeight()` to recalculate the weight again, so the previous change I made to the weight isn't displayed. That doesn't seem like a common set of steps for someone to take, but do you think we should consider it?
<img src="https://user-images.githubusercontent.com/8658164/120630524-d554b380-c45e-11eb-8308-1e8e59904c20.gif" width="300px">

3. This isn't specific to the changes in this PR, but I noticed that we don't validate the weight entered in the `Total package weight` field. I can manually enter an invalid value such as `1..5` or text like `zero` and it's accepted.

pmusolino committed:
fix: validate the total weight in package details

pmusolino committed:
fix: when the total weight is already set, we don't calculate a new total weight

pmusolino commented:
Thank you @rachelmcr!
I fixed the case 1 and the case 3 in this commit: 52d063362dcb7858e7a6f77b8976e5dd13d1ce23
And because on the web when the panel will be closed, they don't re-calculate the total weight if it's already set, I updated our logic: a9c5657d85506f8be69b0fb2acfc0299b7dd9ff0

Ready for another pass! 👍

rachelmcr reviewed (approved):
Code looks good and it tests well. Thanks for making those changes! :shipit:
